shared/modules/types.go

```diff
@@ -1,6 +1,228 @@
-package types
+package modules
 
-// TODO(pocket/issues/163): Move this into utility
+import (
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+// This file contains the minimum shared structures (GenesisState) and the many shared interfaces the modules implement
+// the main purpose of this structure is to ensure the ownership of the
+
+type GenesisState struct {
+	PersistenceGenesisState PersistenceGenesisState `json:"persistenceGenesisState"`
+	ConsensusGenesisState   ConsensusGenesisState   `json:"consensusGenesisState"`
+}
+
+type BaseConfig struct {
+	RootDirectory string `json:"root_directory"`
+	PrivateKey    string `json:"private_key"` // TODO (team) better architecture for key management (keybase, keyfiles, etc.)
+}
+
+type Config struct {
+	Base        *BaseConfig       `json:"base"`
+	Consensus   ConsensusConfig   `json:"consensus"`
+	Utility     UtilityConfig     `json:"utility"`
+	Persistence PersistenceConfig `json:"persistence"`
+	P2P         P2PConfig         `json:"p2p"`
+	Telemetry   TelemetryConfig   `json:"telemetry"`
+}
+
+type ConsensusConfig interface {
+	GetMaxMempoolBytes() uint64
+	GetPaceMakerConfig() PacemakerConfig
+}
+
+type PacemakerConfig interface {
+	SetTimeoutMsec(uint64)
+	GetTimeoutMsec() uint64
+	GetManual() bool
+	GetDebugTimeBetweenStepsMsec() uint64
+}
+
+type PersistenceConfig interface {
+	GetPostgresUrl() string
+	GetNodeSchema() string
+	GetBlockStorePath() string
+}
+
+type P2PConfig interface {
+	GetConsensusPort() uint32
+	GetUseRainTree() bool
+	IsEmptyConnType() bool // TODO (team) make enum
+}
+
+type TelemetryConfig interface {
+	GetEnabled() bool
+	GetAddress() string
+	GetEndpoint() string
+}
+
+type UtilityConfig interface{}
+
+type PersistenceGenesisState interface {
+	GetAccs() []Account
+	GetAccPools() []Account
+	GetApps() []Actor
+	GetVals() []Actor
+	GetFish() []Actor
+	GetNodes() []Actor
+	GetParameters() Params
+}
+
+type ConsensusGenesisState interface {
+	GetGenesisTime() *timestamppb.Timestamp
+	GetChainId() string
+	GetMaxBlockBytes() uint64
+}
+
+type Account interface {
+	GetAddress() string
+	GetAmount() string
+}
+
+type Actor interface {
+	GetAddress() string
+	GetPublicKey() string
+	GetChains() []string
+	GetGenericParam() string
+	GetStakedAmount() string
+	GetPausedHeight() int64
+	GetUnstakingHeight() int64
+	GetOutput() string
+	GetActorTyp() ActorType // TODO (research) this method has to be implemented manually which is a pain
+}
+
+type ActorType interface {
+	String() string
+}
+
+type UnstakingActorI interface {
```

Comment:
Why is `UnstakingActor` the only interface that has the `I` postfix (presuming it stands for Interface)?

Reply:
Because there is a structure inside of `persistence mod` that is called `UnstakingActor` I thought that might be confusing